monai/bundle/config_parser.py

```diff
@@ -266,18 +266,32 @@ def _do_resolve(self, config: Any, id: str = ""):
                 go one level further into the nested structures.
                 Use digits indexing from "0" for list or other strings for dict.
                 For example: ``"xform#5"``, ``"net#channels"``. ``""`` indicates the entire ``self.config``.
+            waiting_list: set of macro replacement ids pending to be resolved.
+                It's used to detect circular references such as:
+                `{"A": {"dep": "%B"}, "B": {"dep": "%A"}}`.
 
         """
+        if waiting_list is None:
+            waiting_list = set()
```

Comment:
If `waiting_list` is actually a list we could look at the previous item in the exception to help the user trace what's happened.